src/test/unit/io/reader_test.cpp

```diff
@@ -1245,6 +1285,63 @@ TEST(io_reader, matrix_lb_constrain_lp) {
   EXPECT_FLOAT_EQ(13.0, a);
 }
 
+TEST(io_reader, sparse_matrix_lb) {
+  auto sparse_pars = stan::test::get_sparse_matrix_reader_pars();
+  auto theta_rows = std::get<0>(sparse_pars);
+  auto theta_cols = std::get<1>(sparse_pars);
+  auto theta = std::get<2>(sparse_pars);
+  auto theta_i = std::get<3>(sparse_pars);
+  auto nonzero_size = std::get<4>(sparse_pars);
+  auto max_row = std::get<5>(sparse_pars);
+  auto max_col = std::get<6>(sparse_pars);
+  stan::io::reader<double> reader(theta, theta_i);
+  double lb = -1.5;
+  Eigen::SparseMatrix<double> y = reader.sparse_matrix_lb(
+      lb, theta_rows, theta_cols, max_row + 1, max_col + 1);
+  for (int i = 0; i < nonzero_size; i++) {
+    EXPECT_FLOAT_EQ(theta[i], y.coeffRef(theta_rows[i], theta_cols[i]));
+  }
+}
```

Comment:
Again, it seems like there might be some abstraction possible here to avoid all this cut-and-paste code.  Especially all this stuff with magic numbers in gets.

In tests, we usually don't need to worry about efficiency, so the constructors, etc. can be abstracted.